control-plane/csi-driver/node/src/node.rs

```diff
@@ -22,11 +22,14 @@ use crate::{
 #[derive(Clone, Debug)]
 pub struct Node {
     pub node_name: String,
+    pub host_name: String,
     pub filesystems: Vec<String>,
 }
 
 const ATTACH_TIMEOUT_INTERVAL: Duration = Duration::from_millis(100);
 const ATTACH_RETRIES: u32 = 100;
+const K8S_HOSTNAME_KEY: &str = "kubernetes.io/hostname";
```

Comment:
Why do we need to host_name again? To keep it simple, could we do without it altogether and just use nodename?

Reply:
We need it for the topology key.

Reply:
Can't the topology simply use node name?

Reply:
Yes, for that we have the `openebs.io/nodename` key, but as per the other engines of openebs, we have both of these keys. So kept it like others.

Reply:
Is it possible to only use `kubernetes.io/hostname` for topology?  Admittedly I'm new to this code but I'm having trouble understanding why both are used from the context in this PR.  I agree it would be better to use the k8s native key wherever possible.

Are there cases where the node name is different from the k8s host name? In my cluster at least they're the same:

Node labels:
```yaml
    kubernetes.io/hostname: aks-storagepool-25014071-vmss000000
    openebs.io/engine: mayastor
    openebs.io/nodename: aks-storagepool-25014071-vmss000000
```
CSINode:
```
  - name: io.openebs.csi-mayastor
    nodeID: aks-storagepool-25014071-vmss000000
    topologyKeys:
    - kubernetes.io/hostname
    - openebs.io/nodename
```

Reply:
Yes, we have found `aws managed clusters` where then nodename, differs from hostname. To handle that case as well we decided to go with this change.

Reply:
Sorry, just a follow up as I'm trying to get my head around this.  I'm not clear why you would need to check both.

It's just a label so it shouldn't really matter as long as you set the values correctly when you register the node.  Previously it registered using the node name rather than the host name:

```rust
segments.insert("kubernetes.io/hostname".to_owned(), self.node_name.clone());
```
Couldn't you just fix this bit to use the host name?

It just seems over-the-top to check both:
```
accessibility requirements: topology map[kubernetes.io/hostname:aks-storagepool-25014071-vmss000001 openebs.io/nodename:aks-storagepool-25014071-vmss000001]
```

Reply:
Previously it used to register with the hostname, rather than the node name. The variable name might be node_name but the value was hostname, if you go through the code. This was causing an issue when hostname differs from nodename.

Reply:
With this change in the PR we are registering with the nodename and not the hostname.

Reply:
Yeah, right. So as dediced we would go with the `nodename` only as that would potentially solve the issue for now, but having said that we need to revisit this later and find a proper way of doing it.